WIP/calculator.rb

```diff
@@ -0,0 +1,95 @@
+class Calculate
+
+  def initialize(**args)
+    @accumulate = args[:accumulate] || args[:accumulate] == nil
+    @discounts = {}
```

Comment:
might be worth to add this as an `attr_reader` so it can be read when needed and we don't need the `return_discounts` method then to return this.